### PR TITLE
AIP-84 Get Event Log

### LINKED ISSUE #43327: AIP-84 Migrate the public endpoint Get Event Log to FastAPI
Migrate the public endpoint `get_event_log` for a single event log item to FastAPI

Part of: https://github.com/apache/airflow/issues/42370

### PR DESCRIPTION
**Pull request by jason810496**
closes: #43327
related: #42370

### CODE REVIEW

File: airflow/api_fastapi/core_api/routes/public/event_logs.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+from airflow.api_fastapi.common.router import AirflowRouter
+from airflow.api_fastapi.core_api.serializers.event_logs import (
+    EventLogResponse,
+)
+from airflow.models import Log
+
+event_logs_router = AirflowRouter(tags=["Event Log"], prefix="/eventLogs")
+
+
+@event_logs_router.get("/{event_log_id}")
```
Comment (pierrejeambrun) [inline airflow/api_fastapi/core_api/routes/public/event_logs.py]: we are missing the errors documentation. (422 is already handled automatically by pydantic, 404 need to be manually added)

File: tests/api_fastapi/core_api/routes/public/test_event_logs.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        log_with_owner_and_task_instance = Log(
+            event=EVENT_WITH_OWNER_AND_TASK_INSTANCE,
+            owner=OWNER,
+            owner_display_name=OWNER_DISPLAY_NAME,
+            task_instance=task_instance,
+        )
+        session.add(normal_log)
+        session.add(log_with_owner)
+        session.add(log_with_task_instance)
+        session.add(log_with_owner_and_task_instance)
```
Comment (pierrejeambrun) [inline tests/api_fastapi/core_api/routes/public/test_event_logs.py]: add_all

File: tests/api_fastapi/core_api/routes/public/test_event_logs.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        session.commit()
+        session.refresh(normal_log)
+        session.refresh(log_with_owner)
+        session.refresh(log_with_task_instance)
+        session.refresh(log_with_owner_and_task_instance)
```
Comment (pierrejeambrun) [inline tests/api_fastapi/core_api/routes/public/test_event_logs.py]: commit should automatically refresh objects.

File: tests/api_fastapi/core_api/routes/public/test_event_logs.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+            return
+
+        resp_json = response.json()
+        when = resp_json["when"]
+        logical_date = resp_json["logical_date"]
+        expected_json = {
+            "event_log_id": event_log_id,
+            "when": when,
+            "dag_id": None,
+            "task_id": None,
```
Comment (pierrejeambrun) [inline tests/api_fastapi/core_api/routes/public/test_event_logs.py]: I don't think `task_id` or `dag_id` should be None for task_instance event.

File: tests/api_fastapi/core_api/routes/public/test_event_logs.py [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+        response = test_client.get(f"/public/eventLogs/{event_log_id}")
+        assert response.status_code == expected_status_code
+        if expected_status_code != 200:
+            return
+
+        resp_json = response.json()
+        when = resp_json["when"]
+        logical_date = resp_json["logical_date"]
+        expected_json = {
+            "event_log_id": event_log_id,
```
Comment (pierrejeambrun) [inline tests/api_fastapi/core_api/routes/public/test_event_logs.py]: This logic is wrong. You construct the `expected_json` based on the test `reponse` that we actually want to assert. ("when", "logical_date" etc...).

File: tests/api_fastapi/core_api/routes/public/test_event_logs.py [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+            "run_id": None,
+            "map_index": event_log.map_index,
+            "try_number": event_log.try_number,
+            "event": None,
+            "logical_date": logical_date,
+            "owner": None,
+            "extra": None,
+        }
+        for key, value in expected_body.items():
+            expected_json[key] = value
```
Comment (pierrejeambrun) [inline tests/api_fastapi/core_api/routes/public/test_event_logs.py]: You can just assert equal two dicts, no need to iterate like this.

**Review by pierrejeambrun — CHANGES_REQUESTED** [review]

**Comment by pierrejeambrun** [comment]
Thanks for the PR. A few suggestions before we can merge.

**Comment by jason810496** [comment]
Thanks for reviewing, @pierrejeambrun. I’ve just resolved the issue.

**Review by pierrejeambrun — APPROVED** [review]

**Pull request merged**